test/integration/link-with-encoding/pages/index.js

```diff
@@ -0,0 +1,15 @@
+import Link from 'next/link'
+
+const Home = () => (
```

Comment:
I think a test should be included that includes `%25`, an escaped slash. This is to ensure the slug is interpreted as `/single/hello%25world` and not `/single/hello/world`. The latter would direct the user to a different page, which is clearly not intended if the slash is escaped.

Reply:
Added more tests.